### DISCUSSION TITLE
About designing graphics like multi-layered as Photoshop or GIMP.

### DISCUSSION THREAD

**Discussion by rageworx** [fire-eggs/YAIV#27] [Ideas]
GUI has changed as you experience in many years.

MacOS early settled efficiant high DPI processing with their cocoa and quartz framework.
Linux is just X11 but theirs many partial distributions, but I believe there's only left with Xserver or not - yes only texts !
Windows ? oh ... it's like hell to me.

I'm using LXDE with openbox for my debian-linux aarch64 on Rockchip RK3399 ARM64 processor, and there's nothing like GPU accelerated drawing because this processor not made for supporting OpenGL ( works with only OpenGL|ES and fine graphics with only Android ).
So in my cases, FLTK is a really great GUI composing tool for really-slow-software-rendered Xserver graphics.

My opinions about great feature of FLTK is I can make graphics like multi-layered in Photoshop or GIMP.
See here my part of designing progress about "self-diagnosis GUI on my embedded linux system for mass-production".

![image](https://user-images.githubusercontent.com/6948225/119596241-c5632100-be19-11eb-9d77-7db441fd2847.png)

I just made each different pre-drawn white line image as PNG... and make it shadowed graphics with fl_imgtk, and draw each different graphics like stacking layers in PS or GIMP.

and you can change white pixels to colored as like below image ( red round-rects and blue lines, these are each png images )

![image](https://user-images.githubusercontent.com/6948225/119596480-31458980-be1a-11eb-8bb8-32e5a9545f10.png)

And Fl_Box on Fl_Box, it's works like layered something - and I call it as overlay.
I can make overlayed layer effect with my Fl_TransparentBox class and it works like this,

![image](https://user-images.githubusercontent.com/6948225/119596620-7d90c980-be1a-11eb-996c-bfc2d636dbe2.png)

You can see a small image in center, it rotates quad size of ellipse ( it's a png image too ) with fl_imgtk::roate().
And final works looks like this.

![image](https://user-images.githubusercontent.com/6948225/119596830-e1b38d80-be1a-11eb-9feb-e91d36d7a469.png)

It was so easy to make GUI design in Windows with MinGW-W64, then easily build in my embedded linux with less changing.
I believe it's attractiveness of FLTK.

![image](https://user-images.githubusercontent.com/6948225/119596946-132c5900-be1b-11eb-8c93-2bc544b8dbb3.png)

You can try windows build ( just for testing what draws ) with this 7z file ( password is "fire-eggs" )
 - https://drive.google.com/file/d/1QSlBkkdI7nnsH-fDYAkL-CFuHNPZMadC/view?usp=sharing

Are these things may helpful to you ?

**Comment by fire-eggs** [comment]
> Are these things may helpful to you ?

I really do appreciate all the thoughts, ideas and suggestions! I might not be able to use all of them, I might be having trouble keeping up :-) but it certainly helps my motivation and engagement!

You've been working with FLTK and image display longer than I, so I anticipate being able to learn a lot  ...   ;-}

**Reply by rageworx** [reply]
> You've been working with FLTK and image display longer than I, so I anticipate being able to learn a lot ... ;-}

You're welcome !
I love FLTK for these features, and you may love too 😁

**Comment by rageworx** [comment]
You can check my forked YAIV version ( don't pull to your repo now ... ) how used Fl_Group as invisible separator, and over layered Fl_TansBox on single Fl_Box(XBox).
Each FLTK Fl_Widgets can stacked to in a Fl_Group ( Fl_Window and Fl_Double_Window too ).
It's very good to design something as drawing layered images in formal graphic editor.
So I love FLTK so much as these features help to make my ideas to be realized for any platform.
